Gemfile

```diff
@@ -4,77 +4,60 @@ gemspec
-
-  # To enable the augeas feature, use this gem.
-  # Note that it is a native gem, so the augeas headers/libs
-  # are neeed.
-  #gem 'ruby-augeas', :group => :development
-end
-
 # override .gemspec deps - may issue warning depending on Bundler version
 gem "facter", *location_for(ENV['FACTER_LOCATION']) if ENV.has_key?('FACTER_LOCATION')
 gem "hiera", *location_for(ENV['HIERA_LOCATION']) if ENV.has_key?('HIERA_LOCATION')
 gem "semantic_puppet", *location_for(ENV['SEMANTIC_PUPPET_LOCATION'] || ["~> 1.0"])
 
-group(:development, :test) do
-  # rake is in .gemspec as a development dependency but cannot
-  # be removed here *yet* due to TravisCI / AppVeyor which call:
-  # bundle install --without development
-  # PUP-7433 describes work necessary to restructure this
-  gem "rake", '~> 12.2.1', :require => false
-  gem "rspec", "~> 3.1", :require => false
-  gem "rspec-its", "~> 1.1", :require => false
-  gem "rspec-collection_matchers", "~> 1.1", :require => false
-  gem "mocha", '~> 1.5.0', :require => false
-  gem "json-schema", "~> 2.0", :require => false
-
-  gem 'rubocop', '~> 0.49', :platforms => [:ruby]
-  gem 'rubocop-i18n', '~> 1.2.0', :platforms => [:ruby]
-
-  gem 'rdoc', "~> 6.0", :platforms => [:ruby]
-  gem 'yard'
-
-  # ronn is used for generating manpages.
-  gem 'ronn', '~> 0.7.3', :platforms => [:ruby]
-
-  gem 'webmock', '~> 1.24'
-  gem 'vcr', '~> 2.9'
-  gem "hiera-eyaml", :require => false
-  gem "hocon", '~> 1.0', :require => false
+group(:features) do
+  gem "hiera-eyaml", require: false
+  gem "hocon", '~> 1.0', require: false
+  gem "msgpack", require: false
+  gem 'rdoc', "~> 6.0", require: false, platforms: [:ruby]
+  # requires native augeas headers/libs
+  # gem 'ruby-augeas', require: false, platforms: [:ruby]
+end
 
-  gem 'memory_profiler', :platforms => [:mri_21, :mri_22, :mri_23, :mri_24, :mri_25]
+group(:test) do
+  gem "json-schema", "~> 2.0", require: false
+  gem "mocha", '~> 1.5.0', require: false
+  gem "rake", '~> 12.2.1', require: false
+  gem "rspec", "~> 3.1", require: false
+  gem "rspec-its", "~> 1.1", require: false
+  gem "rspec-collection_matchers", "~> 1.1", require: false
+  gem 'vcr', '~> 2.9', require: false
+  gem 'webmock', '~> 1.24', require: false
+  gem 'yard', require: false
+
+  gem 'rubocop', '~> 0.49', require: false, platforms: [:ruby]
+  gem 'rubocop-i18n', '~> 1.2.0', require: false, platforms: [:ruby]
 end
 
-group(:development) do
+group(:development, optional: true) do
+  gem 'memory_profiler', require: false, platforms: [:mri]
+  gem 'pry', require: false, platforms: [:ruby]
+  gem "racc", "1.4.9", require: false, platforms: [:ruby]
   if RUBY_PLATFORM != 'java'
-    gem 'ruby-prof', '>= 0.16.0', :require => false
+    gem 'ruby-prof', '>= 0.16.0', require: false
   end
-
-  gem 'gettext-setup', '~> 0.28', :require => false
 end
 
-group(:extra) do
-  gem "puppetlabs_spec_helper", :require => false
-  gem "msgpack", :require => false
+group(:packaging, optional: true) do
```

Comment:
probably need to leave packaging as non-optional so we don't need to change CI pipelines.